keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/ConstraintEvaluator.kt

```diff
@@ -35,3 +39,49 @@ interface ConstraintEvaluator<T : Constraint> {
     targetEnvironment: Environment
   ): Boolean
 }
+
+abstract class StatefulConstraintEvaluator<T : Constraint> : ConstraintEvaluator<T> {
```

Comment:
@luispollo I think this is more in line with what you suggested earlier when I was toying with overloading the `ConstraintEvaluator` interface with stateful bits?